### ISSUE TITLE
Hidden service serving

### ISSUE THREAD

**Issue by AdamISZ** [AdamISZ/CoinSwapCS#32]
A Tor hidden service seems like the most rational/preferred way to set up server side. Adding this to 0.0.1 milestone as it should be in place before publically offering on testnet I think (so it can be tested in practice).

Ideally this will be a default with both clearnet TLS and clearnet non-TLS retained as options.

I investigated this a bit before in the joinmarket context (I believe there's still a `hs_expt` branch on my fork). The package `txtorcon` seemed to work, although I'll have to remember any configuration details. Starting work on this now.

Reminder to self: document setup.

**Comment by AdamISZ** [comment]
#33 changes the design of blockchain access in such a way as to make client-side torsocks wrapping work (see PR comment for details).

**Comment by AdamISZ** [comment]
Want to leave this note here before I forget. Apart from the segfault crash mentioned in the above mentioned PR, which is "fixed" (avoided) by it, another issue that comes up with torsocks is accessing localhost; in torsocks 2.1 and later,  you can set `AllowOutboundLocalhost 1` in `/etc/tor/torsocks.conf` to allow it, else you get an error. I haven't double checked whether this is actually required here, logic dictates that it is (json-rpc access), sadly. In earlier versions of torsocks I don't think this is even possible (? - I'm not sure).

**Comment by AdamISZ** [comment]
Addressed in 4360095da672934ecbe08af593a0c43a74247ce9

But live testing (remote, testnet) now; will need some more tweaks before finishing.

**Comment by AdamISZ** [comment]
Seems to be working OK in tests. First tweak: need to be able to persist onion names, they are currently generated on the fly. This might take a bit of work.

**Comment by AdamISZ** [comment]
Above done in db6898ea6d96a0401867ca8aef4d2e4db517c4f1

This seems to be basically finished, so closing. I note that torsocks 2.1 seems to be required, doesn't seem to be available on ubuntu 14.10, but i think fine on 14.04 and 16.04. Was also tested on debian stretch.
